roles/ceph-osd/tasks/main.yml

```diff
@@ -0,0 +1,109 @@
+---
+- name: ensure raw_multi_journal was chosen
+  fail: msg="please choose an osd scenario"
+  when: ceph.osd_group_name is defined and
+        not ceph_osd.raw_multi_journal
+
+- name: verify devices have been provided
+  fail: msg="please provide devices to your osd scenario"
+  when: ceph.osd_group_name is defined and
+        ceph_osd.raw_multi_journal and
+        ceph_osd.devices is not defined
+
+- name: verify journal devices have been provided
+  fail: msg="please provide journal devices to your osd scenario"
+  when: ceph.osd_group_name is defined and
+        ceph_osd.raw_multi_journal and
+        ceph_osd.raw_journal_devices is not defined
+
+- name: install dependencies
+  apt: pkg={{ item }}
+       state=present
+  with_items: ceph_osd.pkgs
+
+- name: register ceph.keyring
+  slurp: src=/var/lib/ceph/bootstrap-osd/ceph.keyring
+  register: ceph_keyrings
+  run_once: true
+  delegate_to: "{{ groups[ceph.monitor_group_name][0] }}"
+
+- name: write ceph.keyring
+  copy:
+    dest: "{{ ceph_keyrings['source'] }}"
+    content: "{{ ceph_keyrings['content'] | b64decode }}"
+
+- name: check if 'ceph' partition exists on osd devices
+  shell: "parted --script {{ item }} print | egrep -sq '^ 1.*ceph'"
+  with_items: ceph_osd.devices
+  changed_when: false
+  failed_when: false
```

Comment:
Yeah, seems like down below /any/ parted failure could result in sgdisk zapping.